code/mob/living.dm

```diff
@@ -1811,9 +1811,8 @@ var/global/icon/human_static_base_idiocy_bullshit_crap = icon('icons/mob/human.d
 
 	if (!P.was_pointblank && HAS_ATOM_PROPERTY(src, PROP_MOB_REFLECTPROT))
 		var/obj/item/equipped = src.equipped()
-		if (equipped && istype(equipped,/obj/item/sword))
-			var/obj/item/sword/S = equipped
-			S.handle_deflect_visuals(src)
+		var/datum/component/itemblock/saberblock/block_comp = equipped?.GetComponent(/datum/component/itemblock/saberblock)
```

Comment:
`GetComponent` is kinda gross, would prefer just making a signal for this and listening for it in the component.
could make it `COMSIG_ATOM_PROJECTILE_REFLECTED` or something; might be useful for barriers/reflective walls in the future

Reply:
Sure, I had half-formed thoughts of something like that at some point while coding, I think. I will check it out when I have time!